src/Stratis.Bitcoin.Features.Wallet/Controllers/LegacyExtPubKeyConverter.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Linq;
+using NBitcoin;
+using NBitcoin.DataEncoders;
+
+namespace Stratis.Bitcoin.Features.Wallet.Controllers
+{
+    /// <summary>
+    /// Stratis X has a version prefix starting xq5h. Full node has a version prefix xpub for extended public key. 
+    /// 3rd parties like Ledger haven't update yet to the full node version prefix.
+    /// This class provides a way to convert from the legacy format to the current format.
+    /// </summary>
+    /// <remarks>This class can be removed when the 3rd parties update their software.</remarks>
+    public class LegacyExtPubKeyConverter
+    {
+        /// <summary>
+        /// Converts a legacy Stratis format into a current Stratis format Base58 extended public key.
+        /// </summary>
+        /// <param name="extPubKey">The extended public key that may or may not need converting.</param>
+        /// <param name="network">The network to get the version bytes for.</param>
+        /// <returns>The same extended public key if version bytes is already the Stratis full node one,
+        /// or The corrected extended public key if the version bytes was the Stratis X one.</returns>
+        public static string ConvertIfInLegacyStratisFormat(string extPubKey, Network network)
```

Comment:
Please add a unit test

Reply:
It is covered within the Wallet Manager tests, there is no need for explicit unit test